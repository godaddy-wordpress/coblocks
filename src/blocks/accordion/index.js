/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
