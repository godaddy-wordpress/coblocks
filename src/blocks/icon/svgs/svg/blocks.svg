<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>