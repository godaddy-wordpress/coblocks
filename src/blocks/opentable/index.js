/**
 * External dependencies
 */
import { OpentableIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import save from './save';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Block constants.
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'OpenTable', 'coblocks' ),
	/* translators: block description */
	description: __( 'Embed an OpenTable Reservations Widget.', 'coblocks' ),
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'restaurant', 'coblocks' ),
		/* translators: block keyword */
		__( 'reservation', 'coblocks' ),
		/* translators: block keyword */
		__( 'open', 'coblocks' ),
		/* translators: block keyword */
		__( 'table', 'coblocks' ),
	],
	styles: [
		{
			name: 'standard',
			/* translators: block style */
			label: __( 'Standard', 'coblocks' ),
			isDefault: true,
		},
		{
			name: 'tall',
			/* translators: block style */
			label: __( 'Tall', 'coblocks' ),
		},
		{
			name: 'button',
			/* translators: block style */
			label: __( 'Button', 'coblocks' ),
		},
	],
	example: {
		attributes: {
			restaurantIDs: [ { rid: '123456', name: 'Test Restaurant provided by CoBlocks ' } ],
			language: 'en-US',
		},
	},
	attributes,
	edit,
	save,
};

export { name, category, metadata, settings };
