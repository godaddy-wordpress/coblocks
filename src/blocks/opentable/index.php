<?php
/**
 * Handle search function to proxy Opentable request.
 *
 * @package CoBlocks
 */

/**
 * Base Class for proxying Opentable requests.
 */
class Coblocks_Opentable {

	/**
	 * Holds the pase path to our search proxy to send into JS.
	 *
	 * @var string
	 */
	private $search_proxy_path = '/opentable/search/';

	/**
	 * Function constructor.
	 */
	public function __construct() {

		add_action( 'rest_api_init', array( $this, 'api_endpoints' ) );

		add_action( 'enqueue_block_editor_assets', array( $this, 'editor_scripts' ) );

	}

	/**
	 * Registering REST APi proxy.
	 *
	 * @return void
	 */
	public function api_endpoints() {

		register_rest_route(
			COBLOCKS_API_NAMESPACE,
			$this->search_proxy_path . '(?P<query>.*)',
			array(
				'methods'             => \WP_REST_Server::READABLE,
				'permission_callback' => function() {
					// See https://wordpress.org/support/article/roles-and-capabilities/#edit_posts.
					return current_user_can( 'edit_posts' );
				},
				'show_in_index'       => false,
				'args'                => array(
					'query' => array(
						'description'       => 'The query to send opentable.',
						'type'              => 'string',
						'sanitize_callback' => 'sanitize_text_field',
					),
				),
				'callback'            => array( $this, 'api_proxy' ),
			)
		);

	}

	/**
	 * Make the request to the Opentable API and return the result.
	 *
	 * @param \WP_REST_Request $request Contains the rest request object.
	 *
	 * @return \WP_REST_Response
	 */
	public function api_proxy( \WP_REST_Request $request ) {

		$api_resp = wp_remote_get(
			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15&query=' . $request->get_param( 'query' ),
			array(
				// Without this user agent syntax the API doesn't respond at all...
				'user-agent' => 'Mozilla/5.0 Chrome/96 Safari/537',
				'headers'    => array(
					'Accept' => 'application/json',
				),
			)
		);

		if ( is_wp_error( $api_resp ) ) {

			return rest_ensure_response( $api_resp );

		}

		try {

			$body = json_decode( $api_resp['body'], true );

		} catch ( exception $e ) {

			return new \WP_Error( 'parsing_error', 'Error reading body of response.', array( 'status' => 500 ) );

		}

		return rest_ensure_response(
			new \WP_REST_Response(
				$body,
				$api_resp['response']['code']
			)
		);

	}

	/**
	 * Localize script to tell the API proxy path to JavaScript.
	 *
	 * @return void
	 */
	public function editor_scripts() {

		wp_localize_script(
			'coblocks-editor',
			'coblocksOpentable',
			array(
				'searchProxy' => COBLOCKS_API_NAMESPACE . $this->search_proxy_path,
			)
		);

	}

}

new Coblocks_Opentable();
