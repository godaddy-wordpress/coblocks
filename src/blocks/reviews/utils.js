/**
 * External dependencies
 */
import { StarStyles } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies
 */
import { Icon } from '@wordpress/components';

export const renderRating = ( rating, baseRating = 5 ) => {
	const ui = [];

	ui.push( ...Array( rating ).fill( renderRatingIcon( StarStyles.default ) ) );

	if ( baseRating - rating > 0 ) {
		ui.push( ...Array( baseRating - rating ).fill( renderRatingIcon( StarStyles.outlined ) ) );
	}

	return ui;
};

const renderRatingIcon = ( ratingIcon ) => <Icon icon={ ratingIcon } size={ 16 } />;
