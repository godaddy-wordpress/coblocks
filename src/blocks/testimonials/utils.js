/**
 * External dependencies.
 */
import { find } from 'lodash';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import TokenList from '@wordpress/token-list';

export const blockStyles = [
	{
		isDefault: true,
		/* translators: block style */
		label: __( 'Boxy', 'coblocks' ),
		name: 'tall',
	},
	{
		/* translators: block style */
		label: __( 'Conversation', 'coblocks' ),
		name: 'conversation',
	},
	{
		/* translators: block style */
		label: __( 'Horizontal', 'coblocks' ),
		name: 'horizontal',
	},
];

/**
 * Returns the active style from the given className.
 *
 * @param {Array}  styles    Block style variations.
 * @param {string} className Class name
 * @return {Object?} The active style.name.
 */
export const getActiveStyle = ( styles, className ) => {
	for ( const style of new TokenList( className ).values() ) {
		if ( style.indexOf( 'is-style-' ) === -1 ) {
			continue;
		}

		const potentialStyleName = style.substring( 9 );
		const activeStyle = find( styles, { name: potentialStyleName } );

		if ( activeStyle ) {
			return activeStyle.name;
		}
	}

	return find( styles, 'isDefault' ).name;
};
