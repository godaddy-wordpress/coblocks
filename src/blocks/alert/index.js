/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
