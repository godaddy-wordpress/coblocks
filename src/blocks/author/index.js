/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
