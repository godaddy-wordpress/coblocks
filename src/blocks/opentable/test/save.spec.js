/**
 * External dependencies
 */
import { replaceActiveStyle } from '@wordpress/block-editor/build/components/block-styles/utils';
import { createBlock, registerBlockType, serialize } from '@wordpress/blocks';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( 'coblocks/opentable', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should not render when no rid defined', () => {
		block.attributes.restaurantIDs = [ { name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)' } ];
		block.attributes.language = 'en-US';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).not.toContain( 'wp-block-coblocks-opentable' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with preassigned restaurants', () => {
		block.attributes.restaurantIDs = [
			{ name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)', rid: '123456' },
		];
		block.attributes.language = 'en-US';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'iframe__overflow-wrapper' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with style "Tall"', () => {
		block.attributes.restaurantIDs = [
			{ name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)', rid: '123456' },
		];
		block.attributes.language = 'en-US';

		const activeStyle = undefined;
		const newStyle = { name: 'tall' };
		block.attributes.className = replaceActiveStyle( block.attributes.className, activeStyle, newStyle );

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'is-style-tall' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with style "Tall" with multiple', () => {
		block.attributes.restaurantIDs = [
			{ name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)', rid: '123456' },
			{ name: 'Jines Restaurant (Test Restaurant added by CoBlocks)', rid: '246847' },
		];
		block.attributes.language = 'en-US';

		const activeStyle = undefined;
		const newStyle = { name: 'tall' };
		block.attributes.className = replaceActiveStyle( block.attributes.className, activeStyle, newStyle );

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'is-style-tall' );
		expect( serializedBlock ).toContain( 'height:551px' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with style "Button"', () => {
		block.attributes.restaurantIDs = [
			{ name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)', rid: '123456' },
		];
		block.attributes.language = 'en-US';

		const activeStyle = undefined;
		const newStyle = { name: 'button' };
		block.attributes.className = replaceActiveStyle( block.attributes.className, activeStyle, newStyle );

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'is-style-button' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with multiple restaurants selected', () => {
		block.attributes.restaurantIDs = [
			{ name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)', rid: '123456' },
			{ name: 'Jines Restaurant (Test Restaurant added by CoBlocks)', rid: '246847' },
		];
		block.attributes.language = 'en-US';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'type=multi' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with no language selected', () => {
		block.attributes.restaurantIDs = [
			{ name: 'Nuno-Restaurant (Test Restaurant added by CoBlocks)', rid: '123456' },
		];
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'type=standard' );

		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
