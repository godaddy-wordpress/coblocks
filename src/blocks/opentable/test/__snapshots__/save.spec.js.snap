// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/opentable should not render when no rid defined 1`] = `"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\"}],\\"language\\":\\"en-US\\"} /-->"`;

exports[`coblocks/opentable should render with multiple restaurants selected 1`] = `
"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"123456\\"},{\\"name\\":\\"Jines Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"246847\\"}],\\"language\\":\\"en-US\\"} -->
<div class=\\"wp-block-coblocks-opentable iframe__overflow-wrapper\\"><iframe id=\\"opentable-iframe\\" scrolling=\\"no\\" frameborder=\\"0\\" style=\\"width:224px;height:362px;display:block;margin:auto\\" title=\\"open table frame\\" src=\\"//www.opentable.com/widget/reservation/canvas?rid=123456%2C246847&amp;type=multi&amp;domain=com&amp;lang=en-US&amp;theme=standard&amp;overlay=false&amp;insideiframe=true&amp;disablega=true\\"></iframe></div>
<!-- /wp:coblocks/opentable -->"
`;

exports[`coblocks/opentable should render with no language selected 1`] = `
"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"123456\\"}]} -->
<div class=\\"wp-block-coblocks-opentable iframe__overflow-wrapper\\"><iframe id=\\"opentable-iframe\\" scrolling=\\"no\\" frameborder=\\"0\\" style=\\"width:224px;height:302px;display:block;margin:auto\\" title=\\"open table frame\\" src=\\"//www.opentable.com/widget/reservation/canvas?rid=123456&amp;type=standard&amp;domain=com&amp;lang=&amp;theme=standard&amp;overlay=false&amp;insideiframe=true&amp;disablega=true\\"></iframe></div>
<!-- /wp:coblocks/opentable -->"
`;

exports[`coblocks/opentable should render with preassigned restaurants 1`] = `
"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"123456\\"}],\\"language\\":\\"en-US\\"} -->
<div class=\\"wp-block-coblocks-opentable iframe__overflow-wrapper\\"><iframe id=\\"opentable-iframe\\" scrolling=\\"no\\" frameborder=\\"0\\" style=\\"width:224px;height:302px;display:block;margin:auto\\" title=\\"open table frame\\" src=\\"//www.opentable.com/widget/reservation/canvas?rid=123456&amp;type=standard&amp;domain=com&amp;lang=en-US&amp;theme=standard&amp;overlay=false&amp;insideiframe=true&amp;disablega=true\\"></iframe></div>
<!-- /wp:coblocks/opentable -->"
`;

exports[`coblocks/opentable should render with style "Button" 1`] = `
"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"123456\\"}],\\"language\\":\\"en-US\\",\\"className\\":\\"is-style-button\\"} -->
<div class=\\"wp-block-coblocks-opentable iframe__overflow-wrapper is-style-button\\"><iframe id=\\"opentable-iframe\\" frameborder=\\"0\\" scrolling=\\"no\\" style=\\"width:210px;height:116px;display:block;margin:auto\\" title=\\"open table frame\\" src=\\"//www.opentable.com/widget/reservation/canvas?rid=123456&amp;type=button&amp;domain=com&amp;lang=en-US&amp;theme=standard&amp;overlay=false&amp;insideiframe=true&amp;disablega=true\\"></iframe></div>
<!-- /wp:coblocks/opentable -->"
`;

exports[`coblocks/opentable should render with style "Tall" 1`] = `
"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"123456\\"}],\\"language\\":\\"en-US\\",\\"className\\":\\"is-style-tall\\"} -->
<div class=\\"wp-block-coblocks-opentable iframe__overflow-wrapper is-style-tall\\"><iframe id=\\"opentable-iframe\\" frameborder=\\"0\\" scrolling=\\"no\\" style=\\"width:289px;height:491px;display:block;margin:auto\\" title=\\"open table frame\\" src=\\"//www.opentable.com/widget/reservation/canvas?rid=123456&amp;type=standard&amp;domain=com&amp;lang=en-US&amp;theme=tall&amp;overlay=false&amp;insideiframe=true&amp;disablega=true\\"></iframe></div>
<!-- /wp:coblocks/opentable -->"
`;

exports[`coblocks/opentable should render with style "Tall" with multiple 1`] = `
"<!-- wp:coblocks/opentable {\\"restaurantIDs\\":[{\\"name\\":\\"Nuno-Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"123456\\"},{\\"name\\":\\"Jines Restaurant (Test Restaurant added by CoBlocks)\\",\\"rid\\":\\"246847\\"}],\\"language\\":\\"en-US\\",\\"className\\":\\"is-style-tall\\"} -->
<div class=\\"wp-block-coblocks-opentable iframe__overflow-wrapper is-style-tall\\"><iframe id=\\"opentable-iframe\\" frameborder=\\"0\\" scrolling=\\"no\\" style=\\"width:289px;height:551px;display:block;margin:auto\\" title=\\"open table frame\\" src=\\"//www.opentable.com/widget/reservation/canvas?rid=123456%2C246847&amp;type=multi&amp;domain=com&amp;lang=en-US&amp;theme=tall&amp;overlay=false&amp;insideiframe=true&amp;disablega=true\\"></iframe></div>
<!-- /wp:coblocks/opentable -->"
`;
